File: XPile 3-17/Programs/test_refcount.js
// @dr memory = refcount


function allocate_many() {
    let total = 0;
    let i = 0;
    while ((i < 1000)) {
        let s = ("item " + String(i));
        total = (total + str_len(s));
        i = (i + 1);
    }
    return total;
}

function main() {
    let result = allocate_many();
    console.log(result);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_refcount.py
# @dr memory = refcount

from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def allocate_many():
    total = 0
    i = 0
    while (i < 1000):
        s = ('item ' + str(i))
        total = (total + len(s))
        i = (i + 1)
    return total

def main():
    result = allocate_many()
    print(result)
    return 0


if __name__ == "__main__":
    main()
